nikola/metadata_extractors.py

```diff
@@ -0,0 +1,254 @@
+    first_line = 5
+    never = -1
+
+
+class MetaPriority(Enum):
+    """Priority of metadata.
+
+    An extractor is used if and only if the higher-priority extractors returned nothing.
+    """
+
+    override = 1
+    specialized = 2
+    normal = 3
+    fallback = 4
+
+
+class MetaSource(Enum):
+    """Source of metadata."""
+
+    text = 1
+    filename = 2
+
+
+def check_conditions(post, filename: str, conditions: list, config: dict, source_text: str) -> bool:
+    """Check the conditions for a metadata extractor."""
+    for ct, arg in conditions:
+        if any((
+            ct == MetaCondition.config_bool and (arg not in config or (arg in config and not config[arg])),
+            ct == MetaCondition.config_present and arg not in config,
+            ct == MetaCondition.extension and not filename.endswith(arg),
+            ct == MetaCondition.compiler and post.compiler.name != arg,
+            ct == MetaCondition.never
+        )):
+            return False
+        elif ct == MetaCondition.first_line:
+            if not source_text or not source_text.startswith(arg + '\n'):
+                return False
+    return True
+
+
+def classify_extractor(extractor: MetadataExtractor, metadata_extractors_by: dict):
+    """Classify an extractor and add it to the metadata_extractors_by dict."""
+    global DEFAULT_EXTRACTOR
+    if extractor.name == DEFAULT_EXTRACTOR_NAME:
+        DEFAULT_EXTRACTOR = extractor
+    metadata_extractors_by['priority'][extractor.priority].append(extractor)
+    metadata_extractors_by['source'][extractor.source].append(extractor)
+    metadata_extractors_by['name'][extractor.name] = extractor
+    metadata_extractors_by['all'].append(extractor)
+
+
+def load_defaults(site: 'nikola.nikola.Nikola', metadata_extractors_by: dict):
+    """Load default metadata extractors."""
+    for extractor in _default_extractors:
+        extractor.site = site
+        classify_extractor(extractor, metadata_extractors_by)
+
+
+def is_extractor(extractor) -> bool:
+    """Check if a given class is an extractor."""
+    return isinstance(extractor, MetadataExtractor)
+
+
+def default_metadata_extractors_by() -> dict:
+    """Return the default metadata_extractors_by dictionary."""
+    d = {
+        'priority': {},
+        'source': {},
+        'name': {},
+        'all': []
+    }
+
+    for i in MetaPriority:
+        d['priority'][i] = []
+    for i in MetaSource:
+        d['source'][i] = []
+
+    return d
+
+
+def _register_default(extractor: MetadataExtractor) -> MetadataExtractor:
+    """Register a default extractor."""
+    _default_extractors.append(extractor())
+    return extractor
+
+
+@_register_default
+class NikolaMetadata(MetadataExtractor):
+    """Extractor for Nikola-style metadata."""
+
+    name = 'nikola'
+    source = MetaSource.text
+    priority = MetaPriority.normal
+    supports_write = True
+    split_metadata_re = re.compile('\n\n')
+    nikola_re = re.compile('^\s*\.\. (.*?): (.*)')
```

Comment:
Makes sense; I was surprised that it works. I was just confused because here you seemed to use the (deprecated) behavior, while further down you were escaping, and I didn't know if that was intentional or not. But now that's resolved :)